samples/LiveEffect/src/main/cpp/LiveEffectEngine.cpp

```diff
@@ -61,112 +52,26 @@ bool LiveEffectEngine::setAudioApi(oboe::AudioApi api) {
-        closeAllStreams();
-    }
-}
-
-/**
- * Stops and closes the playback and recording streams.
- */
-void LiveEffectEngine::closeAllStreams() {
-    /**
-     * Note: The order of events is important here.
-     * The playback stream must be closed before the recording stream. If the
-     * recording stream were to be closed first the playback stream's
-     * callback may attempt to read from the recording stream
-     * which would cause the app to crash since the recording stream would be
-     * null.
-     */
-
-    if (mPlayStream != nullptr) {
-        closeStream(mPlayStream);  // Calling close will also stop the stream
-        mPlayStream = nullptr;
-    }
-
-    if (mRecordingStream != nullptr) {
-        closeStream(mRecordingStream);
-        mRecordingStream = nullptr;
-    }
-}
-
-/**
- * Creates an audio stream for recording. The audio device used will depend on
- * mRecordingDeviceId.
- * If the value is set to oboe::Unspecified then the default recording device
- * will be used.
- */
-void LiveEffectEngine::openRecordingStream() {
-    // To create a stream we use a stream builder. This allows us to specify all
-    // the parameters for the stream prior to opening it
-    oboe::AudioStreamBuilder builder;
-
-    setupRecordingStreamParameters(&builder);
-
-    // Now that the parameters are set up we can open the stream
-    oboe::Result result = builder.openStream(&mRecordingStream);
-    if (result == oboe::Result::OK && mRecordingStream) {
-        assert(mRecordingStream->getChannelCount() == mInputChannelCount);
-        assert(mRecordingStream->getSampleRate() == mSampleRate);
-        assert(mRecordingStream->getFormat() == oboe::AudioFormat::I16);
-
-        warnIfNotLowLatency(mRecordingStream);
-    } else {
-        LOGE("Failed to create recording stream. Error: %s",
-             oboe::convertToText(result));
-    }
-}
-
-/**
- * Creates an audio stream for playback. The audio device used will depend on
- * mPlaybackDeviceId.
- * If the value is set to oboe::Unspecified then the default playback device
- * will be used.
- */
-void LiveEffectEngine::openPlaybackStream() {
-    oboe::AudioStreamBuilder builder;
-
-    setupPlaybackStreamParameters(&builder);
-    oboe::Result result = builder.openStream(&mPlayStream);
-    if (result == oboe::Result::OK && mPlayStream) {
-        mSampleRate = mPlayStream->getSampleRate();
-
-        assert(mPlayStream->getFormat() == oboe::AudioFormat::I16);
-        assert(mOutputChannelCount == mPlayStream->getChannelCount());
-
-        mSystemStartupFrames =
-            static_cast<uint64_t>(mSampleRate * kSystemWarmupTime);
-        mProcessedFrameCount = 0;
-
-        warnIfNotLowLatency(mPlayStream);
-
-    } else {
-        LOGE("Failed to create playback stream. Error: %s",
-             oboe::convertToText(result));
-    }
+void LiveEffectEngine::bindStreams() {
+    oboe::AudioStreamBuilder inBuilder, outBuilder;
+    setupRecordingStreamParameters(&inBuilder);
```

Comment:
setupRecording probably needs to come after opening the output stream so it can match some properties.